.github/workflows/build-test.yml

```diff
@@ -0,0 +1,45 @@
+name: Snyk Vulnerability Scan
+
+on:
+  push:
+    branches:
+      - main 
+  pull_request:
+    branches:
+      - main  
+
+jobs:
+  snyk-scan:
+    runs-on: ubuntu-latest
+
+    steps:
+   
+      - name: Checkout Code
+        uses: actions/checkout@v3
```

Comment:
i think there is a checkout v4. Can you also check the versions on all of the other actions?

Reply:
updated to v4 and also updated setup-node v4